docs/DeveloperGuide.md

```diff
@@ -29,42 +29,71 @@ Refer to the guide [_Setting up and getting started_](SettingUp.md).
 
 <puml src="diagrams/ArchitectureDiagram.puml" width="280" />
 
-The ***Architecture Diagram*** given above explains the high-level design of the App.
+The ***Architecture Diagram*** given above illustrates the high-level design of the App. 
 
-Given below is a quick overview of main components and how they interact with each other.
+You could see that The bulk of the app's works are done by the UI, Logic, Model and Storage components. Below are the general duties of each component and how they interact with each other.
 
-**Main components of the architecture**
+#### Duties of each component
+[**`UI`**](#ui-component) is responsible for
+* Rendering visuals to the user.
+* Handling and responding to user interactions.
 
-**`Main`** (consisting of classes [`Main`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/Main.java) and [`MainApp`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/MainApp.java)) is in charge of the app launch and shut down.
```

Comment:
I think we should still include this as it is the component that is running the application.

Reply:
its included below